src/clj_kondo/impl/analyzer.clj

```diff
@@ -359,7 +359,7 @@
               arg-list (sexpr arg-vec)
               arity (analyze-arity arg-list)
               ret (cond-> {:arg-bindings (dissoc arg-bindings :analyzed)
-                           :arity arity
```

Comment:
I'm not sure this is fine. `parsed-bodies` doesn't really map back to the original expression which i need for location info, the best alternative i could think of would be doing `(reduce (fn [] ...) (mapv list parsed-bodies bodies)` which feels a bit awkward as well.

Reply:
Two things:

- I think pointing to the vector would be fine too.
- An alternative to the meta idea is to assoc `arg-vec` into the arity map and pull it out, only when you actually need the location in case of an error.

Reply:
adapted to `arg-vec` ... i think it would be a bit confusing to just use the metadata this way on data that's not a direct parsing result. i kind of like the idea that derived data can point to a source location via metadata as well. if this were to be introduced it would probably make more sense in a larger scope and also documented, otherwise it'll be more confusing than helpful... do you think that would be worth pursuing?

Reply:
>  i think it would be a bit confusing to just use the metadata this way on data that's not a direct parsing result. 

Are you referring to something which is the case _after_ your latest commit? I don't understand.

Reply:
no, i was referring to my initial solution that transfers the metadata of the function body to `arity`. now i'm just passing around the `arg-vector`.

Reply:
Sometimes results aren't maps and then the only solution (without changing a lot of code) is to attach it to metadata. I don't have a strong opinion on this approach, but I think we should not introduce changes without a good reason (problem statement, issue).